src/Connection/Http1Connection.php

```diff
@@ -372,18 +379,23 @@ private function readResponse(
                             // to resolve promise with headers.
                             $chunk = null;
 
-                            do {
-                                /** @noinspection CallableParameterUseCaseInTypeContextInspection */
-                                $parser->parse($chunk);
-                                /** @noinspection NotOptimalIfConditionsInspection */
-                                if ($parser->isComplete()) {
-                                    break;
-                                }
-
-                                if (!$backpressure instanceof Success) {
-                                    yield $this->withCancellation($backpressure, $bodyCancellationToken);
-                                }
-                            } while (null !== $chunk = yield $this->socket->read());
+                            try {
+                                do {
+                                    /** @noinspection CallableParameterUseCaseInTypeContextInspection */
+                                    $parser->parse($chunk);
+                                    /** @noinspection NotOptimalIfConditionsInspection */
+                                    if ($parser->isComplete()) {
+                                        break;
+                                    }
+
+                                    if (!$backpressure instanceof Success) {
+                                        yield $this->withCancellation($backpressure, $bodyCancellationToken);
+                                    }
+                                } while (null !== $chunk = yield Promise\timeout($this->socket->read(), $timeout));
+                            } catch (PromiseTimeoutException $e) {
+                                $this->close();
+                                throw new TimeoutException('Receiving the response timed out due to inactivity', 0, $e);
```

Comment:
This should probably include the configured timeout value like all other timeout error messages.